app/dashboard/notifications.py

```diff
@@ -406,117 +407,92 @@ def build_github_notification(bounty, event_name, profile_pairs=None):
-        else:
-            msg = f"{status_header}__Workers have applied to start work__.\n\n"
-
-        msg += f"\nThese users each claimed they can complete the work by {from_now}. " \
-               "Please review their questions below:\n\n"
-
-        for i, interest in enumerate(interested, start=1):
-
-            profile_link = f"[{interest.profile.handle}]({interest.profile.url})"
-            action = "started work"
-            if interest.pending:
-                action = 'applied to start work'
-                action += f" _(Funders only: [approve worker]({bounty.approve_worker_url(interest.profile.handle)})"
-                action += f" | [reject worker]({bounty.reject_worker_url(interest.profile.handle)}))_"
-            if not interest.pending and approval_required:
-                action = 'been approved to start work'
-
-            show_dibs = interested.count() > 1 and bounty.project_type == 'traditional'
-            dibs = f" ({get_ordinal_repr(i)} dibs)" if show_dibs else ""
-
-            msg += f"\n{i}. {profile_link} has {action}{dibs}. "
-
-            issue_message = interest.issue_message.strip()
-            if issue_message:
-                msg += f"\t\n * Q: " \
-                       f"{issue_message}"
-            msg += "\n\n"
+        pending_interest = bounty.interested.filter(pending=True).select_related('profile').all()
+        if bounty.permission_type == 'approval':
+            for interest in pending_interest:
+               interested_context.append(
+                       SimpleNamespace(
+                           profile = interest.profile,
+                           approve_link = bounty.approve_worker_url(interest.profile.handle),
+                           reject_link = bounty.reject_worker_url(interest.profile.handle),
+                           issue_comment = interest.issue_message
+                        )
+                )
+            context = {
+                'bounty': bounty,
+                'interested': interested_context
+            }
+            
+            msg['pending'] = render_to_string(template_path + 'express_interest.md', context)
+        started_work = bounty.interested.filter(pending=False).select_related('profile').all()
+        if started_work.count():
+            for interest in started_work:
+                started_context.append(
+                        SimpleNamespace(
+                            profile = interest.profile,
+                            issue_comment = interest.issue_message
+                        )
+                )
+            context = {
+                'bounty': bounty,
+                'started_work': started_context}
+            msg['started'] = render_to_string(template_path + 'start_work.md', context)
 
+    elif event_name == 'stop_work':
+        started_work = [interest.profile for interest in bounty.interested.filter(pending=False).select_related('profile').all()]
```

Comment:
We'll want to revisit the way these queries are performed and how the data is established for iteration/rendering.

Reply:
@mbeacom hmmmm should it iterate only once and store in a variable?

Reply:
@mbeacom or should we be selecting only the necessary fields out?

Reply:
Made this for DRYing and selecting only the fields required in the queries for iteration/rendering. This applies to everything but pending_interest
`def build_related_profile_pairs(queryset_all):
    # *args:
    #   a 1: QuerySet:  objects to get related profiles for
    queryset_all = queryset_all.select_related('profile').only('profile__handle')
    return queryset_all`